### PR TITLE
DEVOPS-590

### PR DESCRIPTION
**Pull request by iris-ersc**
- cmake: fix target SDK
- mongoose: fix package splitting
- quirc: Add Findquirc.cmake and add patch
- seasocks: Add patches
- dlib: skip linking of some libraries

### CODE REVIEW

**Comment by irisPHSC** [comment]
is there a special reason for lower-casing the mongoos and quirc cmake project/target?

**Review by irisPHSC — APPROVED** [review]

File: recipes-core/seasocks/seasocks-1.4.4/0002-add-ipv6-support-to-seasocks.patch [hunk 1]
```diff
@@ -0,0 +1,110 @@
+From 319cfe25b3ef290a63b8fa734619a16d22ca1946 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Christian=20M=C3=BCller?= <christian.multer@irisgmbh.de>
+Date: Wed, 22 Apr 2020 08:45:22 +0200
+Subject: [PATCH] add ipv6 support to seasocks
```
Comment (linkjumper) [inline recipes-core/seasocks/seasocks-1.4.4/0002-add-ipv6-support-to-seasocks.patch:4]: Let's drop this patch. We don't plan to use ipv6 internally.

**Comment by iris-ersc** [comment]
> is there a special reason for lower-casing the mongoos and quirc cmake project/target?

CMake is case-sensitive and changing the capitalization here is a lot easier than changing a batch of CMakeLists.txt in multiple other projects.

File: recipes-core/seasocks/seasocks-1.4.4/0002-add-ipv6-support-to-seasocks.patch (hunk 1 above)
Comment (iris-ersc) [reply recipes-core/seasocks/seasocks-1.4.4/0002-add-ipv6-support-to-seasocks.patch:4]: True! We should (in general) check if patches are still necessary, to stay as close as possible to upstream.

But I would argue against doing it now. The scope of DEVOPS-590 was only to switch to dynamic linking with minimal changes on the platform applications.

**Review by Eximmius — APPROVED** [review]

**Pull request merged**